Projects/Project10 Names to Faces/Project10 Names to Faces/ViewController.swift

```diff
@@ -7,56 +7,132 @@
+        addFromCameraButton?.isEnabled = false
+        addFromPhotosButton?.isEnabled = false
+        
         if UIImagePickerController.isSourceTypeAvailable(.camera) {
-            let cameraButton = UIBarButtonItem(barButtonSystemItem: .camera, target: self, action: #selector(addNewPersonByCamera))
-            navigationItem.leftBarButtonItems = [addButton, cameraButton]
+            navigationItem.rightBarButtonItems = [addFromPhotosButton!, addFromCameraButton!]
         } else {
-            navigationItem.leftBarButtonItem = addButton
+            navigationItem.rightBarButtonItem = addFromPhotosButton
         }
+        
+        navigationItem.leftBarButtonItem = authenticateButton
     }
     
-    @objc func addNewPerson() {
-        let picker = UIImagePickerController()
-        picker.allowsEditing = true
-        picker.delegate = self
-        present(picker, animated: true)
+    @objc func lockApp() {
+        title = "Locked"
+        authenticateButton?.isEnabled = true
+        addFromCameraButton?.isEnabled = false
+        addFromPhotosButton?.isEnabled = false
+        people = []
+        self.collectionView.reloadData()
     }
     
-    @objc func addNewPersonByCamera() {
-        let picker = UIImagePickerController()
-        picker.sourceType = .camera
-        picker.allowsEditing = true
-        picker.delegate = self
-        present(picker, animated: true)
+    @objc func authenticateTapped() {
+        let context = LAContext()
+        var error: NSError?
+        if context.canEvaluatePolicy(.deviceOwnerAuthentication, error: &error) {
+            let reason = "Identify yourself with touchID"
+            context.evaluatePolicy(.deviceOwnerAuthentication, localizedReason: reason) { [weak self] (success, error) in
+                if success {
+                    // authentication might not happen on the main thread so use GCD
+                    DispatchQueue.main.async {
+                        self?.loadPeople()
+                        self?.authenticateButton?.isEnabled = false
+                        self?.addFromCameraButton?.isEnabled = true
+                        self?.addFromPhotosButton?.isEnabled = true
+                        self?.title = "Unlocked"
+                    }
+                } else {
+                    // alert failed to authenticate using biometrics
+                    let ac = UIAlertController(title: "FaceID Failed", message: "Biometric faceID failed in recognizing your face.", preferredStyle: .alert)
+                    ac.addAction(UIAlertAction(title: "Dismiss", style: .default))
+                    self?.present(ac, animated: true)
+                }
+            }
+        } else {
+            // alert device does not have biometry authenticating capabilities
+            let ac = UIAlertController(title: "Biometry Scanning Unavailble", message: "You're device is not configured for biometric authentication.", preferredStyle: .alert)
```

Comment:
Try not abbreviating in swift 
https://swift.org/documentation/api-design-guidelines/#promote-clear-usage